cannot connect on windows

Hi guys, i can run the qt client on windows fine.

I'm trying to connect using the python client now.

I set the NVIM_LISTEN_ADDRESS variable but i receive an error
```python
vim = neovim.attach('socket', path=os.environ['NVIM_LISTEN_ADDRESS'])
```
```
File "C:\Python34\lib\asyncio\events.py", line 330, in create_unix_connection
raise NotImplementedError
```
I've read that others can get it working.. is this a new bug or am i missing a step?

The same behaviour happens with python2.

after attching to neovim `vim.command_output("echo $NVIM_LISTEN_ADDRESS")` works.

However running the tests.
```
    from common import vim, cleanup
common.py:13: in <module>
    vim = neovim.attach('socket', path=os.environ['NVIM_LISTEN_ADDRESS'])
C:\Python35\lib\os.py:683: in __getitem__
    raise KeyError(key) from None
E   KeyError: 'NVIM_LISTEN_ADDRESS'
```

The problem is present on python2 as well.

** Failing tests **
* test_command
* test_command_output

```
FAIL: test_vim.test_command
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\python34\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\Users\timeyyy\Documents\git\python-client\test\test_vim.py", line 24, in test_command
    eq(open(fname).read(), 'testing\npython\napi\n')
nose.proxy.AssertionError: '\ntesting\npython\napi\n' != 'testing\npython\napi\n'
-------------------- >> begin captured logging << --------------------
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 323, 'vim_command', ('call BeforeEachTest()',)]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01C\xabvim_command\x91\xb5call BeforeEachTest()''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 323, None, None]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, None
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 324, 'vim_get_tabpages', ()]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01D\xb0vim_get_tabpages\x90''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 324, None, [ExtType(code=2, data=b'\xcd\x01\xae')]]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, [ExtType(code=2, data=b'\xcd\x01\xae')]
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 325, 'vim_get_windows', ()]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01E\xafvim_get_windows\x90''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 325, None, [ExtType(code=1, data=b'\xcd\x01\xaf')]]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, [ExtType(code=1, data=b'\xcd\x01\xaf')]
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 326, 'vim_get_buffers', ()]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01F\xafvim_get_buffers\x90''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 326, None, [ExtType(code=0, data=b'\xcd\x01\xb0')]]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, [ExtType(code=0, data=b'\xcd\x01\xb0')]
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 327, 'vim_command', ('new',)]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01G\xabvim_command\x91\xa3new''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 327, None, None]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, None
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 328, 'vim_command', ('edit C:\\Users\\timeyyy\\.babun\\cygwin\\tmp\\tmpdyw5jtcg',)]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01H\xabvim_command\x91\xd93edit C:\\Users\\timeyyy\\.babun\\cygwin\\tmp\\tmpdyw5jtcg''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 328, None, None]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, None
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 329, 'vim_input', ('\r',)]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01I\xa9vim_input\x91\xa1\r''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 329, None, 1]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, 1
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 330, 'vim_command', ('normal itesting\npython\napi',)]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01J\xabvim_command\x91\xbanormal itesting\npython\napi''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 330, None, None]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, None
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: sent [0, 331, 'vim_command', ('w',)]
neovim.msgpack_rpc.event_loop.base: DEBUG: Sending 'b'\x94\x00\xcd\x01K\xabvim_command\x91\xa1w''
neovim.msgpack_rpc.event_loop.base: DEBUG: Entering event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: received message: [1, 331, None, None]
neovim.msgpack_rpc.async_session: DEBUG: received response: None, None
neovim.msgpack_rpc.event_loop.base: DEBUG: Stopped event loop
neovim.msgpack_rpc.msgpack_stream: DEBUG: waiting for message...
neovim.msgpack_rpc.msgpack_stream: DEBUG: unpacker needs more data...
neovim.msgpack_rpc.event_loop.base: DEBUG: Exited event loop
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 44 tests in 1.441s

FAILED (errors=1, failures=1)
Exception ignored in: <bound method _ProactorDuplexPipeTransport.__del__ of <_ProactorDuplexPipeTransport closing fd=656 read=<_OverlappedFuture cancelled>>>
Traceback (most recent call last):
  File "c:\python34\lib\asyncio\proactor_events.py", line 86, in __del__
  File "c:\python34\lib\asyncio\proactor_events.py", line 74, in close
  File "c:\python34\lib\asyncio\base_events.py", line 427, in call_soon
  File "c:\python34\lib\asyncio\base_events.py", line 436, in _call_soon
  File "c:\python34\lib\asyncio\base_events.py", line 265, in _check_closed
RuntimeError: Event loop is closed

```

pytest was able to capture a little more info about the second test 
```
    @with_setup(setup=cleanup)
    def test_chdir():
        pwd = vim.eval('getcwd()')
>       vim.chdir('/')

Users\timeyyy\Documents\git\python-client\test\test_vim.py:60:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
users\timeyyy\documents\git\python-client\neovim\api\nvim.py:163: in chdir
    return self._session.request('vim_change_directory', dir_path)
users\timeyyy\documents\git\python-client\neovim\api\common.py:230: in request
    return walk(self._in, self._session.request(name, *args, **kwargs),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <neovim.msgpack_rpc.session.Session object at 0x0333AD50>
method = 'vim_change_directory', args = ('/',), kwargs = {}, async = False
v = [[0, 'Failed to change directory'], None]
err = [0, 'Failed to change directory'], rv = None

    def request(self, method, *args, **kwargs):
        """Send a msgpack-rpc request and block until as response is received.

            If the event loop is running, this method must have been called by a
            request or notification handler running on a greenlet. In that case,
            send the quest and yield to the parent greenlet until a response is
            available.

            When the event loop is not running, it will perform a blocking request
            like this:
            - Send the request
            - Run the loop until the response is available
            - Put requests/notifications received while waiting into a queue

            If the `async` flag is present and True, a asynchronous notification is
            sent instead. This will never block, and the return value or error is
            ignored.
            """
        async = kwargs.get('async', False)
        if async:
            self._async_session.notify(method, args)
            return

        if self._is_running:
            v = self._yielding_request(method, args)
        else:
            v = self._blocking_request(method, args)
        if not v:
            # EOF
            raise IOError('EOF')
        err, rv = v
        if err:
            info("'Received error: %s", err)
>           raise self.error_wrapper(err)
E           NvimError: Failed to change directory

```

> Now there was one :) nose.proxy.AssertionError: '\ntesting\npython\napi\n' != 'testing\npython\napi\n'

Mmmm, should probably leave this one for another issue. I was able to skip the error by removing the vim.input('\r') line, so it would seem the input is affecting the other command. But in any case the `os.unlink()` is also problematic in Windows because the file is still open in Vim.

@equalsraf the `chdir` test passes if you change to C:\ :

```
vim.chdir('C:\\')
eq(vim.eval('getcwd()'), 'C:\\')
```

I'm left with the same:

```
nose.proxy.AssertionError: '\ntesting\npython\napi\n' != 'testing\npython\napi\n
```

--- a/neovim/msgpack_rpc/event_loop/asyncio.py
+++ b/neovim/msgpack_rpc/event_loop/asyncio.py
@@ -80,7 +80,10 @@ class AsyncioEventLoop(BaseEventLoop, asyncio.Protocol,
         self._loop.run_until_complete(coroutine)
 
     def _connect_socket(self, path):
-        coroutine = self._loop.create_unix_connection(self._fact, path)
+        if os.name == 'nt':
+            coroutine = self._loop.create_pipe_connection(self._fact, path)
+        else:
+            coroutine = self._loop.create_unix_connection(self._fact, path)
         self._loop.run_until_complete(coroutine)
 
     def _connect_stdio(self):
@@ -111,6 +114,11 @@ class AsyncioEventLoop(BaseEventLoop, asyncio.Protocol,
         self._loop.call_soon_threadsafe(fn)
 
     def _setup_signals(self, signals):
+        if os.name == 'nt':
+            # add_signal_handler is not supported in win32
+            self._signals = []
+            return
+
         self._signals = list(signals)
         for signum in self._signals:
             self._loop.add_signal_handler(signum, self._on_signal, signum)

